### ISSUE TITLE
Hw 4 Uniqueness Issue

### ISSUE THREAD

**Issue by brett1479** [davidrosenberg/mlcourse#74]
In https://github.com/davidrosenberg/mlcourse/blob/gh-pages/Homework/hw4.pdf in question 3.4 I don't see why alpha is uniquely determined.  Perhaps the question should say that you are finding an expression for a valid alpha, but not the only one.

**Comment by brett1479** [comment]
Also, don't a few of the steps disappear if we use the identity:
<a href="https://www.codecogs.com/eqnedit.php?latex=(X^TX&plus;\lambda&space;I)^{-1}X^T&space;=&space;X^T(XX^T&plus;\lambda&space;I)^{-1}" target="_blank"><img src="https://latex.codecogs.com/gif.latex?(X^TX&plus;\lambda&space;I)^{-1}X^T&space;=&space;X^T(XX^T&plus;\lambda&space;I)^{-1}" title="(X^TX+\lambda I)^{-1}X^T = X^T(XX^T+\lambda I)^{-1}" /></a>

**Comment by davidrosenberg** [comment]
Yes that's a good point.  For our definition of alpha, we could have added anything in the nullspace of X^T.

And yeah you can use that matrix inversion identity to drop steps.  That may actually be more useful.  I've never liked this development, actually.  It's cute, in that every step is really elementary.  But it's also a little bit too clever, imo.
